src/lightning_app/runners/cloud.py

```diff
@@ -96,8 +97,13 @@ def dispatch(
 
         print(f"The name of the app is: {app_config.name}")
 
-        work_reqs: List[V1Work] = []
+        secret_names_to_ids = _names_to_ids(self.secrets.values())
+        env_vars_from_secrets = [V1EnvVar(name=k, from_secret=secret_names_to_ids[v]) for k, v in self.secrets.items()]
+
         v1_env_vars = [V1EnvVar(name=k, value=v) for k, v in self.env_vars.items()]
+        v1_env_vars.extend(env_vars_from_secrets)
```

Comment:
Q: The encryption does not happen locally, only once it reaches the backend, correct?

Reply:
I don't fully understand the workflow here. It seems that in _names_to_ids we fetch the user's (?) secrets, then append new ones and send the whole thing back.